lib/typeform_data/requestor.rb

```diff
@@ -12,47 +14,61 @@ def self.get(config, endpoint, params = nil)
       "/v#{config.api_version}/#{endpoint}"
     end
 
-    # rubocop:disable Metrics/MethodLength
     # @return TypeformData::ApiResponse
     private_class_method def self.request(config, method_class, path, input_params = {})
       params = input_params.dup
       params[:key] = config.api_key
 
-      Utils.retry_with_exponential_backoff([TypeformData::UnexpectedError]) do
+      begin
+        Utils.retry_with_exponential_backoff(RETRY_EXCEPTIONS, max_retries: 3) do
+          request_and_validate_response(config, method_class, path, params)
+        end
+      rescue *DOMAIN_EXCEPTIONS
+        raise
 
-        response = Net::HTTP.new(config.host, config.port).tap { |http|
-          http.use_ssl = true
+      # Why are we rescuing StandardError? See http://stackoverflow.com/a/11802674/1067145
+      rescue StandardError => error
```

Comment:
I think we should only rescue `StandardError` around the actual `Net::HTTP#request` call, since otherwise it might catch bugs in `typeform_data`'s code.